src/test/ReactTestUtils.js

```diff
@@ -80,14 +80,35 @@ function findAllInRenderedTreeInternal(inst, test) {
  * @lends ReactTestUtils
  */
 var ReactTestUtils = {
-  renderIntoDocument: function(instance) {
+  renderIntoDocument: function(element) {
     var div = document.createElement('div');
     // None of our tests actually require attaching the container to the
     // DOM, and doing so creates a mess that we rely on test isolation to
     // clean up, so we're going to stop honoring the name of this method
     // (and probably rename it eventually) if no problems arise.
     // document.documentElement.appendChild(div);
-    return ReactDOM.render(instance, div);
+    return ReactDOM.render(element, div);
+  },
+
+  renderIntoDocumentAsync: function(element) {
+    var promise = new Promise(function(fulfill, reject) {
+      var div = document.createElement('div');
+      // None of our tests actually require attaching the container to the
+      // DOM, and doing so creates a mess that we rely on test isolation to
+      // clean up, so we're going to stop honoring the name of this method
+      // (and probably rename it eventually) if no problems arise.
```

Comment:
We will also eventually need a way to render into a container asynchronously.

What about `ReactTestUtils.renderAsync()` which optionally accepts the second argument (`container`) or renders into a detached node if no such container exists.

`renderDetached` seems verbose, since 99.99% of the time, you don't care that the node is detached.  You just want to render something.

Was just an idea.  I truly don't have a strong preference here.  Just let me know in a more definitive way when everyone is done bikeshedding and then I'll do the global replace.